src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -331,18 +331,16 @@ private function getContainerDefinitionDocument(Definition $definition, $id = nu
 
         $serviceXML->setAttribute('class', $definition->getClass());
 
-        if (method_exists($definition, 'getFactoryMethod')) {
-            if ($definition->getFactoryClass(false)) {
-                $serviceXML->setAttribute('factory-class', $definition->getFactoryClass(false));
-            }
+        if ($definition->getFactoryClass(false)) {
+            $serviceXML->setAttribute('factory-class', $definition->getFactoryClass(false));
+        }
 
-            if ($definition->getFactoryService(false)) {
-                $serviceXML->setAttribute('factory-service', $definition->getFactoryService(false));
-            }
+        if ($definition->getFactoryService(false)) {
+            $serviceXML->setAttribute('factory-service', $definition->getFactoryService(false));
+        }
 
-            if ($definition->getFactoryMethod(false)) {
-                $serviceXML->setAttribute('factory-method', $definition->getFactoryMethod(false));
-            }
+        if ($definition->getFactoryMethod(false)) {
```

Comment:
It looks like it has never been supported. The DI dependency was bumped to 2.8 (with no explicit support for 3.0) in 73f0ee24136d28a5fc525c0706700f49f36a3eb4 without obvious reasons and stayed as is since 2 years.